sql-composer/src/parser.rs

```diff
@@ -491,234 +505,222 @@ named!(
+
+    Ok((span, vec![Sql::Ending(e)]))
+}
+
+pub fn sql_ending_item(span: Span) -> IResult<Span, ParsedItem<SqlEnding>> {
+    let (span, ending) = tag(";")(span)?;
+    let (span, _) = multispace0(span)?;
+
+    let item = ParsedItem::from_span(
+        SqlEnding::new(ending.fragment.to_string())
+            .expect("SqlEnding::new() failed unexpectedly from parse_sql_end parser"),
+        ending,
+        None,
     )
-);
+    .expect("expected Ok from ParsedItem::from_span in parse_sql_end");
 
-#[cfg(feature = "composer-serde")]
-pub fn bind_value_text(
-    span: Span,
-    ) -> IResult<Span, SerdeValue> {
+    Ok((span, item))
+}
 
+#[cfg(feature = "composer-serde")]
+pub fn bind_value_text(span: Span) -> IResult<Span, SerdeValue> {
     let (span, _) = one_of("'")(span)?;
-    let (span, found) = take_while1(
-        |c:char| match c {
-            '\'' => false,
-            ']'  => false,
-            _    => true,
-        })(span)?;
+    let (span, found) = take_while1(|c: char| match c {
+        '\'' => false,
+        ']' => false,
+        _ => true,
+    })(span)?;
     let (span, _) = one_of("'")(span)?;
     let (span, _) = multispace0(span)?;
     let (span, _) = check_bind_value_ending(span)?;
-    Ok((span,
-        SerdeValue(Value::String(found.fragment.to_string()))))
+
+    Ok((span, SerdeValue(Value::String(found.fragment.to_string()))))
 }
 
 #[cfg(feature = "composer-serde")]
-pub fn bind_value_integer(
-    span: Span,
-    ) -> IResult<Span, SerdeValue> {
-
+pub fn bind_value_integer(span: Span) -> IResult<Span, SerdeValue> {
     let (span, found) = digit1(span)?;
     let (span, _) = multispace0(span)?;
     let (span, _) = check_bind_value_ending(span)?;
-    Ok((span,
-        SerdeValue(Value::I64(i64::from_str(&found.fragment).expect("unable to parse integer found by bind_value_integer")))))
+
+    Ok((
+        span,
+        SerdeValue(Value::I64(
+            i64::from_str(&found.fragment)
+                .expect("unable to parse integer found by bind_value_integer"),
+        )),
+    ))
 }
 
 #[cfg(feature = "composer-serde")]
-pub fn bind_value_real(
-    span: Span,
-    ) -> IResult<Span, SerdeValue> {
-
+pub fn bind_value_real(span: Span) -> IResult<Span, SerdeValue> {
     let (span, value) = double(span)?;
     let (span, _) = multispace0(span)?;
     let (span, _) = check_bind_value_ending(span)?;
     let (span, _) = multispace0(span)?;
-    Ok((span,
-        SerdeValue(Value::F64(value))))
+
+    Ok((span, SerdeValue(Value::F64(value))))
+}
+
+pub fn ending(span: Span) -> IResult<Span, Span> {
+    //! Checks for end of input
+    //!
+    //! succeeds if input_len is zero
+    //! Returns an Eof Error if input_len is non-zero.
+    //!   Returns Error rather than Failure to indicate
+    //!   "I didn't match" vs "match can not succeed"
+    if span.input_len() == 0 {
+        Ok((span, span))
+    } else {
+        Err(nom::Err::Error((span, NomErrorKind::Eof)))
+    }
```

Comment:
Is this clearer as if/else or as a match.  you know I love matches.  :fire: It is a line shorter and the two return values happen to align.

```
    match span.input_len() {
        0 => Ok((span, span)),
        _ => Err(nom::Err::Error((span, NomErrorKind::Eof)))
    }

```

Reply:
Yeah, these matches are definitely growing on me,